File: SWEA/D1/2056. 연월일 달력/연월일 달력.py
days = {1: 31, 2: 28, 3:31, 4:30, 5:31, 6: 30, 7: 31, 8: 31, 9: 30, 10: 31, 11: 30, 12: 31}
T = int(input())

for test_case in range(1, T+1):
    result = ''
    full = str(input())
    y = full[0:4]
    m = full[4:6]
    d = full[6:8]

    if 1 <= int(m) <= 12 and 1 <= int(d) <= days[int(m)]:
        result = y + '/' + m + '/' + d
    else:
        result += '-1'

    print(f'#{test_case} {result}')

File: SWEA/D1/2056. 연월일 달력/연월일 달력.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    int T;
    cin >> T;
    int days[13] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    
    for(int tc = 1 ; tc <= T ; tc++){
        string date;
        cin >> date;
        
        int y = stoi(date.substr(0, 4));
        int m = stoi(date.substr(4, 2));
        int d = stoi(date.substr(6, 2));
        
        string result;
        if(1 <= m && m <= 12 && 1 <= d && d <= days[m]) result = date.substr(0, 4) + "/" + date.substr(4, 2) + "/" + date.substr(6, 2);
        else result = "-1";
        
        cout << "#" << tc << " " << result << endl;
    }
    return 0;
}